src/CliScreen.php

```diff
@@ -20,47 +21,51 @@ public function __construct()
      */
     public function clear()
     {
-        fwrite(STDOUT, shell_exec('tput reset'));
-
-        // Calculate and print FPS
-        fwrite(STDOUT, shell_exec('tput cup 0 2'));
-        fwrite(STDOUT, 'FPS: ' . $this->getFps());
+        CLI\Erase::screen();
+        CLI\Output::string("FPS: {$this->getFps()}", 1, 4);
     }
 
     /**
      * @param Drawable $entity
      */
     public function draw(Drawable $entity)
     {
-        $coordinates = $entity->getCoordinates();
-
-        foreach (range($coordinates['y'], $coordinates['y'] + $coordinates['h'] - 1) as $y) {
-            fwrite(STDOUT, shell_exec('tput cup ' . $y . ' ' . $coordinates['x']));
+        $c = $entity->getCoordinates();
 
-            foreach (range($coordinates['x'], $coordinates['x'] + $coordinates['w'] - 1) as $x) {
-                fwrite(STDOUT, $entity->getDrawingCharacter());
-            }
+        // Translate coordinates to \CLI\Graphics format
+        if ($c['w'] == 1 && $c['h'] == 1) {
+            $x1 = $c['x'] + 1;
+            $y1 = $c['y'] + 1;
+            $x2 = $x1;
+            $y2 = $y1;
+        } else {
+            $x1 = $c['x'] + 1;
+            $y1 = $c['y'] + 1;
+            $x2 = $c['x'] + $c['w'] - 1;
+            $y2 = $c['y'] + $c['h'];
         }
 
+        CLI\Graphics::line($x1, $y1, $x2, $y2, [$entity->getDrawingCharacter()]);
+
         // Set the cursor to the top left of the window to prevent having
         // a weird cursor flicker follow the last drawn element
-        fwrite(STDOUT, shell_exec('tput cup 0 0'));
+        CLI\Cursor::hide();
```

Comment:
For me it had a glitch if I didn't hide it. It could also be moved but hide was an obvious choice.